packages/smooth_app/lib/pages/preferences/user_preferences_languages_list.dart

```diff
@@ -0,0 +1,417 @@
+  'gl': 'galego',
+  'gu': 'ગુજરાતી',
+  'ha': 'હૌસા',
+  'he': 'עִברִית',
+  'hi': 'हिन्दी',
+  'hr': 'Hrvatski',
+  'ht': 'ayisyen',
+  'hu': 'Magyar',
+  'hy': 'հայերեն',
+  'id': 'bahasa Indonesia',
+  'ii': 'Sichuan Yi',
+  'is': 'íslenskur',
+  'it': 'Italiano',
+  'iu': 'Inuktitut',
+  'ja': '日本語',
+  'jv': 'basa jawa',
+  'ka': 'ქართული',
+  'kk': 'қазақ',
+  'km': 'ខ្មែរ',
+  'kn': 'ಕನ್ನಡ',
+  'ko': '한국인',
+  'ku': 'Kurdî',
+  'kw': 'Cornish',
+  'ky': 'Кыргызча',
+  'la': 'latīnum',
+  'lb': 'lëtzebuergesch',
+  'lo': 'ພາສາລາວ',
+  'lt': 'lietuvių',
+  'lv': 'latviski',
+  'mg': 'Malagasy',
+  'mi': 'മലയാളം',
+  'mn': 'Монгол',
+  'mr': 'मराठी',
+  'ms': 'Melayu',
+  'mt': 'Malti',
+  'my': 'မြန်မာဘာသာ',
+  'nb': 'Norsk bokmål',
+  'ne': 'नेपाली',
+  'nl': 'Nederlands',
+  'nn': 'Norsk nynorsk',
+  'no': 'norsk',
+  'nr': 'South Ndebele',
+  'oc': 'Occitan',
+  'pa': 'Panjabi',
+  'pl': 'Polski',
+  'pt': 'Português',
+  'qu': 'Runasimi',
+  'rm': 'Romansh',
+  'ro': 'Română',
+  'ru': 'Русский',
+  'sa': 'संस्कृत',
+  'sc': 'Sardinian',
+  'sd': 'سنڌي',
+  'sg': 'Sango',
+  'si': 'සිංහල',
+  'sk': 'Slovenčina',
+  'sl': 'Slovenščina',
+  'sn': 'Shona',
+  'so': 'Soomaali',
+  'sq': 'shqiptare',
+  'sr': 'Српски',
+  'ss': 'Swati',
+  'st': 'Southern Sotho',
+  'sv': 'svenska',
+  'sw': 'kiswahili',
+  'ta': 'தமிழ்',
+  'te': 'తెలుగు',
+  'tg': 'тоҷикӣ',
+  'th': 'ไทย',
+  'ti': 'ትግሪኛ',
+  'tl': 'Tagalog',
+  'tn': 'Setswana',
+  'tr': 'Türk',
+  'ts': 'Tsonga',
+  'tt': 'Татар',
+  'tw': 'Twi',
+  'ty': 'Tahitian',
+  'ug': 'ئۇيغۇر',
+  'uk': 'Українська',
+  'ur': 'اردو',
+  'uz': "o'zbek",
+  've': 'Venda',
+  'vi': 'Tiếng Việt',
+  'wa': 'Walloon',
+  'wo': 'Wolof',
+  'xh': 'isiXhosa',
+  'yi': 'יידיש',
+  'yo': 'Yoruba',
+  'zh': '中文',
+  'zu': 'Zulu',
+};
+
+/*
+  Args:
+    language Name: The language Name.
+
+    * Non of the language supported has name smaller then 3 characters
+    If the language length is 2 then it means the language name is language code.
+
+  Returns:
+    List of language code and language name in that language
+*/
+
+List<String> getLanguageCodeLangName(String languageName) {
+  // Incase of language code is provided
+  if (languageName.length == 2) {
+    if (supportedLanguages.containsKey(languageName)) {
+      return <String>[languageName, supportedLanguages[languageName]!];
+    } else {
+      // Incase of language code is not supported
+      return <String>[languageName, languageName];
+    }
+  }
+
+  final Map<String, List<String>> languagesNameTranslation =
+      <String, List<String>>{
+    'Afar': <String>['aa', 'Afar'],
+    'Afrikaans': <String>['af', 'Afrikaans'],
+    'Akan': <String>['ak', 'Akan'],
+    'Amharic': <String>['am', 'አማርኛ'],
+    'Arabic': <String>['ar', 'عربى'],
+    'Assamese': <String>['as', 'অসমীয়া'],
+    'Azerbaijani': <String>['az', 'Azərbaycan'],
+    'Belarusian': <String>['be', 'беларускi'],
+    'Bulgarian': <String>['bg', 'български'],
```

Comment:
No way. You cannot just copy code or data 3 times.
Instead, something like 
```dart
  OpenFoodFactsLanguage.BULGARIAN:
    TranslatedLanguage('Bulgarian', 'български'),
```

Reply:
Thanks for the suggestion.
It looks much cleaner and more efficient.